ADR/0046-common-task-runner-image.md

```diff
@@ -0,0 +1,131 @@
+  code in unnatural ways and sharing some data between Task steps. Inexperienced
+  Tekton users may not even think of this approach or know how to achieve it.
+* Increased compute resource requirements for the Task. The total resource requirements
+  for a Task are not the *maximum* of its steps' resource requirements, they are
+  the *sum* (see [Compute Resources in Tekton][compute-resources-in-tekton]).
+* Reduced size limit of the results that the task can return (unless the Tekton
+  installation enables [Results from sidecar logs][results-from-sidecar-logs]).
+
+### Konflux users and custom Tasks
+
+The Enterprise Contract team has developed the Trusted Artifacts concept to enable
+Konflux users to add custom Tasks to the pipelines without compromising the
+trustworthiness of the build.
+
+But Konflux users face the same difficulties described above (made worse by the
+fact that they don't tend to have much Tekton experience). The initial hurdle of
+finding/building the right container image for what they want to do may be too high.
+
+## Decision
+
+Build and maintain a common "Task Runner" image.
+
+The image must:
+
+* Include all the tools commonly needed by Konflux build tasks.
+* Build and release via Konflux, hermetically if possible.
+* Document the list of installed tools and their versions, similar to how GitHub
+  documents the [software installed in their runner images][github-runner-software].
+  * The list of tools is a public interface, both Konflux devs and Konflux users
+    can depend on it.
+* Use proper semver versioning. The deletion of a tool, or a change in the major
+  version of a tool, is a breaking change and must result in a major version change
+  for the Task Runner image.
+
+Gradually deprecate all the current tool-oriented container images and replace
+their usage with the common Task Runner image.
+
+The Task Runner image does not replace the more specialized use-case-oriented images,
+but they can use it as a base image if desirable.
+
+## Consequences
+
+The maintenance of container images needed for Tasks moves to one place. The total
+number of rebuilds needed due to CVEs stays the same but is not scattered across
+tool-container repos anymore.
+
+Tasks get easier to write because all the tools you need are available in the same
+image. For both Konflux devs and Konflux users.
+
+Tasks have lower resource requirements because it's no longer necessary to split
+them into steps.
+
+The Task Runner image is larger than any of the individual images used by the Tasks
+at present. But it's much smaller than all the individual images combined. And
+because Pipelines pull each image at most once, this is a win.
+
+By no longer relying on a Tekton-specific feature (steps), most Tasks become nothing
+more than a bash script wrapped in some YAML. It enables a saner approach to authoring
+Tasks. Write a bash script that works on your machine, wrap it in a bunch of YAML,
```

Comment:
Although I agree with this, it seems to be contrary to this [conversation](https://github.com/konflux-ci/architecture/pull/217/files#r1846022425). Are you saying that it's ok to have a mixture of random scripts in the runner image?

Reply:
Exceptions would still need to exist.

`use-trusted-artifact` / `create-trusted-artifact` steps would still be needed (until TA becomes a standalone tool and the TA-generation tool takes that into account, if ever)

The `build-source-image` task would still use the image tailor-built for that one task, until the script becomes a standalone tool (if ever).

I'll rephrase this to make it clearer

Reply:
Softened the wording, added an example where we would still want to use steps